shared/modules/factory.go

```diff
@@ -4,23 +4,29 @@ package modules
 // a variadic `ModuleOption` argument(s) and returns a `Module`and an error.
 type ModuleFactoryWithOptions FactoryWithOptions[Module, ModuleOption]
 
-// FactoryWithConfig implements a `#Create()` factory method which takes a
-// required "config" argument of type K and returns a value of type T and an error.
+// Factory implements a `#Create()` factory method which takes a bus and returns
+// a value of type T and an error.
+type Factory[T interface{}] interface {
```

Comment:
Would this be better declared as an any instead of an interface? I think they're functionally identical, but any seems more appropriate in the context of generics.

Reply:
Thanks for the reminder, I keep forgetting thats a thing! :raised_hands:

Reply:
Thanks for the reminder, I keep forgetting thats a thing! :raised_hands: